src/state.rs

```diff
@@ -9,17 +12,17 @@ use crate::domain::services::Services;
 #[derive(Clone)]
 pub struct State {
     pub services: Services,
+    pub db_pool: Arc<PgPool>,
 }
 
 impl State {
     pub fn new() -> Result<Self> {
         let db_pool = make_db_pool()?;
         let db_pool = Arc::new(db_pool);
 
-        let services = Services::new(db_pool)?;
+        let services = Services::new(db_pool.clone())?;
```

Comment:
Pleas use associated function `clone` from `Arc` here:

```suggestion
        let services = Services::new(Arc::clone(&db_pool))?;
```